Assets/HoloToolkit/Speech/Scripts/TextToSpeechManager.cs

```diff
@@ -0,0 +1,334 @@
+
+                        // Create buffer
+                        byte[] buffer = new byte[(int)size];
+
+                        // Get input stream and the size of the original stream
+                        using (var inputStream = speechStream.GetInputStreamAt(0))
+                        {
+                            // Close the original speech stream to free up memory
+                            speechStream.Dispose();
+
+                            // Create a new data reader off the input stream
+                            using (var dataReader = new DataReader(inputStream))
+                            {
+                                // Load all bytes into the reader
+                                await dataReader.LoadAsync((uint)size);
+
+                                // Copy from reader into buffer
+                                dataReader.ReadBytes(buffer);
+                            }
+                        }
+
+                        // Convert raw WAV data into Unity audio data
+                        int sampleCount = 0;
+                        int frequency = 0;
+                        var unityData = ToUnityAudio(buffer, out sampleCount, out frequency);
+
+                        // The remainder must be done back on Unity's main thread
+                        UnityEngine.WSA.Application.InvokeOnAppThread(() =>
+                        {
+                            // Convert to an audio clip
+                            var clip = ToClip("Speech", unityData, sampleCount, frequency);
+
+                            // Set the source on the audio clip
+                            audioSource.clip = clip;
+
+                            // Play audio
+                            audioSource.Play();
+                        }, false);
+                    });
+                }
+                catch (Exception ex)
+                {
+                    Debug.LogErrorFormat("Speech generation problem: \"{0}\"", ex.Message);
+                }
+            }
+            else
+            {
+                Debug.LogErrorFormat("Speech not initialized. \"{0}\"", text);
+            }
+        }
+        #endif
+
+        private void StartSpeech()
+        {
+            try
+            {
+                if (audioSource == null)
+                {
+                    Debug.LogError("An AudioSource is required and should be assigned to 'Audio Source' in the inspector.");
+                }
+                else
+                { 
+                    #if WINDOWS_UWP
+                    synthesizer = new SpeechSynthesizer();
+                    #endif
+                }
+            }
+            catch (Exception ex)
+            {
+                Debug.LogError("Could not start Speech Synthesis");
+                Debug.LogException(ex);
+            }
+        }
+
+        // MonoBehaviour Methods
+        void Start()
+        {
+            // Start speech
+            StartSpeech();
```

Comment:
I mean, we can move the StartSpeech code to Start, but I feel like we're splitting hairs a little. Especially since that doesn't affect the public interface. If you want me to move the .cs file to the Utilities folder I can do that too, but I would like to ask what else we need to take care of in order to get this pull request approved. The original pull request was submitted 28 days ago and I'm starting to feel like it's never going to make it in.

I'm jbienz in the GAL. Maybe we could setup an internal call to discuss any remaining issues and get them all taken care of and accepted?

Thanks!